### PR TITLE
Remove dead modules

### PR DESCRIPTION
**Pull request by WilliamJamieson**
This PR removes some dead modules:

- `roman_datamodels.utils`
- `roman_datamodels.mktest`

These modules appear to be untested, unused, duplicate other functionality, and essentially abandoned.

**Checklist**
- [x] Added entry in `CHANGES.rst` under the corresponding subsection
- [x] updated relevant tests
- [x] updated relevant documentation
- [x] Passed romancal regression testing on Jenkins / PLWishMaster. Link: https://plwishmaster.stsci.edu:8081/job/RT/job/Roman-Developers-Pull-Requests/236/ (passes with spacetelescope/romancal#719 changes)

### CODE REVIEW

File: tests/test_models.py [hunk 1]
```diff
@@ -191,6 +191,24 @@ def test_opening_association_ref(tmp_path):
...
+        (False, {"foo": "bar"}),
+        (False, "foo"),
+    ],
+)
+def test_is_association(expected, asn_data):
+    """
+    Test the is_association function.
+    """
+
+    assert datamodels.AssociationsModel.is_association(asn_data) is expected
```
Comment (WilliamJamieson) [inline tests/test_models.py:209]: Actually, test the `is_association` (it was never tested before).

File: src/roman_datamodels/datamodels.py [hunk 2]
```diff
@@ -375,7 +375,16 @@ class RampFitOutputModel(DataModel):
...
+
+    @classmethod
+    def is_association(cls, asn_data):
+        """
+        Test if an object is an association by checking for required fields
+        """
+        if isinstance(asn_data, dict):
+            if "asn_id" in asn_data and "asn_pool" in asn_data:
+                return True
+        return False
```
[the remark this answers is not in the document]
Comment (mairanteodoro) [reply src/roman_datamodels/datamodels.py:387]: @WilliamJamieson I'm OK with moving it out of `roman_datamodels`.

**Comment by WilliamJamieson** [comment]
Close/reopen due to messing up branch pushes

File: docs/roman_datamodels/datamodels/developer_api.rst
Comment (mairanteodoro) [inline docs/roman_datamodels/datamodels/developer_api.rst:1]: Are we adding these to the `romancal` docs?

Comment (WilliamJamieson) [reply docs/roman_datamodels/datamodels/developer_api.rst:1]: No, they were linked to in the `roman_datamodels` docs accidentally.

**Review by PaulHuwe — APPROVED** [review]
LGTM modulo @mairanteodoro's approval.

**Review by mairanteodoro — APPROVED** [review]

**Pull request merged**